src/compiler/checker.ts

```diff
@@ -9257,6 +9316,12 @@ namespace ts {
                         if (!bodyDeclaration) {
                             bodyDeclaration = node;
                         }
+
+                        // abstract functions cannot have an implementation.
+                        // Extra checks are to avoid reporting multiple errors relating to the "abstractness" of the node.
+                        if (currentNodeFlags & NodeFlags.Abstract && node.kind === SyntaxKind.MethodDeclaration && !isConstructor) {

```

Comment:
The cleanest spot to move the check to is checkMethodDeclaration, so I've done that.